### DISCUSSION TITLE
Is there a way to use a client extension to define a method on all results with $allModels?

### DISCUSSION THREAD

**Discussion by nickyhajal** [prisma/prisma#18072] [Q&A]
Is there a way to use a client extension to define a method on all results with $allModels? For example, to add save() in this example to all models:

![image](https://user-images.githubusercontent.com/1079625/221005553-fda13f9e-b122-416d-9305-f0ea450948a2.png)

$allModels auto completes when adding a custom result extension, so it seems like it may be possible:

![image](https://user-images.githubusercontent.com/1079625/221005601-ab18cfc4-353f-41c4-a643-fda6e7c0437a.png)

**Comment by millsp** [comment]
Yes, that is possible, but the API will not let you define a `needs` property because it will apply to anything.

**Reply by nickyhajal** [reply]
What are the ramifications of not being able to define a needs property? Simply no proper types, or `compute` will not receive any values?

Also, is there any way to receive a reference to the parent model dynamically so that it would be possible to access the proper update method? (`prisma.<model>.update`)

Thanks!

**Reply by millsp** [reply]
You will receive the data in `compute` for `$allModels` but it will be of type `unknown`. And no, currently it is not possible to reference to reference the model that is tied to a specific `compute` call. This is interesting. Would you mid creating a detailed  feature request with some more details about your use-case?

**Comment by broisnischal** [comment]
```ts
result: {
      $allModels: {
        save: {
          compute(data: any) {
            console.log(data);
            return () =>
              prisma.<model>.update({ where: { id: data.id }, data: data });
          },
        },
      },
    },
  });
});
```

wait how?

**Reply by broisnischal** [reply]
I couldn't figure out how to update!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
